src/core/base.jl

```diff
@@ -6,16 +6,58 @@ export
     run_generic_model, build_generic_model, solve_generic_model
 
 ""
-@compat abstract type AbstractPowerModel end
-
-""
 @compat abstract type AbstractPowerFormulation end
 
 ""
 @compat abstract type AbstractConicPowerFormulation <: AbstractPowerFormulation end
 
-""
-type GenericPowerModel{T<:AbstractPowerFormulation} <: AbstractPowerModel
+"""
+```
+type GenericPowerModel{T<:AbstractPowerFormulation}
+    model::JuMP.Model
+    data::Dict{String,Any} 
+    setting::Dict{String,Any}
+    solution::Dict{String,Any}
+    ref::Dict{Symbol,Any} # reference data
+end
+```
+where
+
+* `data` is the original data, usually from reading in a `.json` or `.m` (patpower) file
+* `setting` usually looks something like `Dict("output" => Dict("line_flows" => true))`, and
+* `ref` is a `Dict` with the following keys:
```

Comment:
I think this level of detail about ref, should be left to the function that populates the ref data structure.  Here is an overview of what this is for,

Ref is a place to store commonly used pre-computed data from of the data dictionary.  The three primary uses are,
1) convert dictionary lookups from "string" => object to int => object, so that the type of ids inside objects is the same as the lookup.
2) filter out components that are deactivated in the data
3) store some system-wide values that need to be computed globally.  Such as identifying the reference bus or computing `off_angmin`.